src/libraries/System.Net.NameResolution/tests/FunctionalTests/GetHostAddressesTest.cs

```diff
@@ -174,19 +174,33 @@ public async Task DnsGetHostAddresses_PreCancelledToken_Throws()
         [OuterLoop]
         [Fact]
         [ActiveIssue("https://github.com/dotnet/runtime/issues/43816")] // Race condition outlined below.
-        [ActiveIssue("https://github.com/dotnet/runtime/issues/33378", TestPlatforms.AnyUnix)] // Cancellation of an outstanding getaddrinfo is not supported on *nix.
         public async Task DnsGetHostAddresses_PostCancelledToken_Throws()
         {
-            using var cts = new CancellationTokenSource();
+            const int numberOfRequests = 100;
+
+            using CancellationTokenSource cts = new();
+            List<Task> tasks = new(capacity: numberOfRequests);
 
-            Task task = Dns.GetHostAddressesAsync(TestSettings.UncachedHost, cts.Token);
+            for (int i = 0; i < numberOfRequests; ++i)
+            {
+                Task task = Dns.GetHostAddressesAsync(TestSettings.UncachedHost, cts.Token);
```

Comment:
See https://github.com/dotnet/runtime/issues/43816#issuecomment-748088532
If this is consider reliable enough, so that issue is fixed with this change.

Reply:
I'll let @wfurt comment, but this seems like at some point it's going to fail spuriously in CI.

Reply:
It seems like the TestSettings.UncachedHost is already used twice and the second use _could_ be cached - certainly in case of systemd resolver. If we want to improve chance of failure, I think each round should use unique name.  And the test was already problematic in CI. 

We could also accept the reality that cancellation is best-effort and timing sensitive and verify that _if_ task failed, OperationCanceledException is only one reason. 

We could also use SkipTestException. That would not change the test run much but we could at least see reports over time.

Reply:
So we could do this strategy:

1. update the tests here to use SkipTestException, so it won't break CI but we still get some data for the cancellation-tests
1. provide a proper / more robust solution when adressing #43816

Reply:
62f05fe25098f96ccba82dc238174ed2a94f4be6 adds `SkipTestException`. OK as proposed in the comment before?